momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -1115,393 +1115,458 @@ public CompletableFuture<CacheSortedSetRemoveElementsResponse> sortedSetRemoveEl
   }
 
   /**
-   * Concatenates values to the back of the list.
+   * Adds the given elements to the back of a list.
    *
-   * @param cacheName Name of the cache to store the item in
-   * @param listName The list in which the value is to be added.
-   * @param values The elements to add to the list.
+   * @param cacheName The cache containing the list.
+   * @param listName The list in which to add the elements.
+   * @param elements The elements to add to the list.
    * @param truncateFrontToSize If the list exceeds this length, remove excess from the front of the
    *     list. Must be positive.
-   * @param ttl Time to Live for the item in Cache. This ttl takes precedence over the TTL used when
-   *     building a cache client {@link CacheClient#builder(CredentialProvider, Configuration,
-   *     Duration)}
-   * @return Future containing the result of the list concatenate back operation.
+   * @param ttl TTL for the set in cache. This TTL takes precedence over the TTL used when
+   *     initializing a cache client. Defaults to client TTL.
+   * @return Future containing the result of the list concatenate back operation: {@link
+   *     CacheListConcatenateBackResponse.Success} or {@link
+   *     CacheListConcatenateBackResponse.Error}.
    */
   public CompletableFuture<CacheListConcatenateBackResponse> listConcatenateBack(
-      String cacheName,
-      String listName,
-      List<String> values,
-      int truncateFrontToSize,
+      @Nonnull String cacheName,
```

Comment:
@cprice404 What do you think of the way these look with fully annotated arguments? Nonnull is helpful for generating good warnings for the user and it would look strange to not annotate the Nullable arguments.